frame/dvm-dynamic-fee/src/lib.rs

```diff
@@ -81,12 +81,16 @@ pub mod pallet {
 	#[pallet::call]
 	impl<T: Config> Pallet<T> {
 		/// Set the target gas price.
-		#[pallet::weight(T::DbWeight::get().writes(1))]
-		fn note_min_gas_price_target(
+		#[pallet::weight((T::DbWeight::get().writes(1), DispatchClass::Mandatory))]
+		pub fn note_min_gas_price_target(
 			origin: OriginFor<T>,
 			target: U256,
 		) -> DispatchResultWithPostInfo {
 			ensure_none(origin)?;
+			assert!(
```

Comment:
`assert!` looks dangerous, will it cause the runtime panic?

Reply:
No. We use `<TargetMinGasPrice<T>>::take()` in `on_finalize`. 

1. More discuss here https://github.com/paritytech/frontier/pull/435#discussion_r680966368
2. Substrate timestamp usage: https://github.com/paritytech/substrate/blob/master/frame/timestamp/src/lib.rs#L196

Reply:
What will happen to the block producers if they intent to insert multiple target gas?

Especially when do runtime upgrade, on_runtime_upgrade is after last runtime's final block's finalize and before next runtime's first block, all block producers might do same thing to note, if all of them are stalled, the chain will be stalled.

I worry that this is some indentical behavior for all clients.

Timestamp pallet do not happen might just because there were no runtime upgrade including such changes in production network.

We need further learn of client behavior or testing of this.

Reply:
In case we forget, we need to do migration to clean/remove `<TargetMinGasPrice<T>>` for runtime upgrade.